### DISCUSSION TITLE
Targetting version 0.9

### DISCUSSION THREAD

**Discussion by brunoherbelin** [brunoherbelin/vimix#140] [Announcements]
The versioning of **vimix** is organic, following the stream of ideas I have and requests that [users post](https://github.com/brunoherbelin/vimix/discussions). The evolution of [previous versions](https://github.com/brunoherbelin/vimix/wiki/Future-plans) and [list of releases](https://github.com/brunoherbelin/vimix/releases) shows the past. Here I welcome your input to give the target for the future !

I see two main directions:
 1. Towards video mapping : multiple output areas in geometry view, control multiple windows, link to video mapping software, etc.
 2. Towards video sequencer : time-control of events, cyclic animation of sources, recording of user actions and replay, etc.

But maybe you see something else? Please feel free to give you opinion on the priority for you and ideas you have for vimix! I am not saying that this will be done, nor when it could be done, but guarantee that I will take your input into account for establishing the list of priorities for the upcoming version 0.9 !

**Comment by MoorPoor** [comment]
Fore sure vimix needs (and has already) the best of both of two worlds ;)

Concerning Video sequencer and control of contend chaitaigne opened up a lot for me. Sensors Inputs, audio reactivity and so on, all stuff I was missing in vimix. But actually there is no need to implement it anymore as a combination of this two softwares is a dream. 

Mapping of contend on the other hand needs to be done in vimix from my point of view. Also I tend to say, that it is fundamental to get the video material where it needs to be, no madder, if you re mapping or VJing or just producing. 

So even thought I want to raise my hand for both ways, I tend to 1. as it is just much more useful for my work.

Thank you so much for your work! And your general openness for use suggestions! Big benefit for artidts and friends of open software

**Comment by lupin3rd** [comment]
I agree with @MoorPoor. The priority could be to work on video mapping.
A quick solution could be to support existing video mapping software.
(For example use Splash software by piping through shmdata.)
In the long term it would be nice to have basic video mapping functionality within Vimix, so that you don't need another software unless you need advanced functionality.

**Comment by brunoherbelin** [comment]
Well, lucky for you, vimix already supports shmdata sharing with Splash;

![Screenshot from 2024-05-18 14-27-41](https://github.com/brunoherbelin/vimix/assets/60393189/76f08fda-584c-41f8-89fb-c02f78e82e96)

It works out-of-the-box with the flatpak packages of vimix and Splash. Select `shmdatasink` in vimix Settings of SHM plugin, and enter as shm link file `.shm_vimix0' (with full path for home directory) in Splash

**Reply by brunoherbelin** [reply]
Update: Improved support for `shmdata` in latest commits and fixed compilation instructions in [README](https://github.com/brunoherbelin/vimix/blob/master/README.md).

**Reply by lupin3rd** [reply]
I tested... and works very fine with last beta!
In README you must change:
cmake -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_BUILD_TYPE=Release -DWITH_PYTHON=0 -DWITH_SDCRASH=0 -DWITH_SDFLOW=0 ../shmdata-build
with:
cmake -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_BUILD_TYPE=Release -DWITH_PYTHON=0 -DWITH_SDCRASH=0 -DWITH_SDFLOW=0 ../shmdata

**Comment by zcbernard** [comment]
Good morning
The desires regarding the facilities of Vimix depend on how we conceive of live editing and improvisation. Everyone wants them in view of their practice. For my part, playing video in conjunction with an improvising musician, I quite like the "cut" edit. That is to say moving from one session to another instantly and over very short periods of time. To put it better: a session takes place, we "play" it, we wish to switch to another, would only be for a few seconds, then resume the previous one where it had been stopped.... Intersperse a third... while keeping the planned course of a session with possible or already prepared actions. I will be able to take the image of an instrument with its sound facets.
Bernard

**Comment by a-1219** [comment]
Hi, I would say video mapping is one of the main lacking functionalities in the open source landscape atm. Splash has always been somewhat clunky for me. I tried using ossia score as a VJ software and it's pretty capable, but not as smooth as vimix, although it has really interesting controlling and compositing capabilities. It also has a video mapping approach that worked better for me than figuring out splash.

**Comment by brunoherbelin** [comment]
Getting closer to 0.9 with new features for video mapping!
Two novelties you can now try in Beta version (da8f1947feb54f8076d5e4bacca390cb947a5325 at the time of this post)

## Geometry view

Define canvases in the Geometry view as portions of the mix output. You can have multiple canvases for mixing your sources.

NB: the session still has a global resolution fixed, because this is necessary to establish the base size of all framebuffers. The resolution can be set to a custom size and as large as your computer graphics card can handle!

<img width="1867" height="1008" alt="20260128220451947_vmixcapture" src="https://github.com/user-attachments/assets/7af94e12-6116-42c8-91dc-38a2c4e4d7ba" />

## Displays view

Places as many canvases as necessary in the monitors area, with free placement and distortion (as sources in Geometry view). No more windows for output, only ON/OFF projection on a monitor (fullscreen).

<img width="1867" height="1008" alt="20260128220448130_vmixcapture" src="https://github.com/user-attachments/assets/d7a3457c-7351-4267-aea7-0001829c5ed8" />

## Examples

1. Mosaic of canvases to project with multiple surfaces with projection adjustments.

<img width="1113" height="725" alt="Screenshot From 2026-01-28 22-10-49" src="https://github.com/user-attachments/assets/664b4c3d-5407-4193-a64f-b34f925adbf6" />

<img width="1162" height="737" alt="Screenshot From 2026-01-28 22-12-28" src="https://github.com/user-attachments/assets/7ba8cf19-e440-4365-b69c-630b4b9ab72f" />



2. Two distinct canvases for multiple mixing areas, projected on two projectors

<img width="1867" height="1008" alt="20260128221704774_vmixcapture" src="https://github.com/user-attachments/assets/c02f9c2d-59e5-41f3-947f-834f0ef70318" />

<img width="1867" height="1008" alt="20260128221858008_vmixcapture" src="https://github.com/user-attachments/assets/d4a372b9-9f2e-4a9f-b5c8-6f3d1a0364b6" />


## Thanks for testing !

**Reply by zcbernard** [reply]
It's impeccable, truly excellent. In the spirit of Vimix (as I envision it), the tool is merely the reason for the visual (and sound...) production. The way it considers time and space, primarily as a space for play, and not as an obligation to fulfill (literally or figuratively), is fascinating. "To show must go on," of course, but Vimix raises the question of "how to do things differently from the norm." Recognition for creating this tool.
regards
BZtruc

**Comment by zcbernard** [comment]
Hi,
An idea just popped into my head!

If it were possible to record gestures applied to an image or video, it would then allow those gestures to be applied to another image or video later. By gesture, I mean moving an object in or out of the circle, changing its geometry, or its position on the screen...
Another idea: the central disk split vertically into two halves, each half assigned to a screen output (if there are two screens, of course).

Regarding the integration of Vimix with other software (sound, controls, etc.), I have no idea and no desire to do so. Having worked with complex sound systems (sequencer connected to a synthesizer, plus MIDI controls), I don't want to encounter that again in my work. Vimix is ​​already very rich in possibilities.

To be continued.

regatrds

**Comment by zcbernard** [comment]
Hello,
Regarding the modifications to Vimix, would it be possible to add an extra level of functionality to the "Save As" feature?

That is, to have the option to "Save Project As," which would save all sessions and offer the option to move and/or copy source files (images, videos, etc.) to the project folder. This type of saving exists in audio software like REAPER or Soundplant. It's useful when source files are scattered across various folders, and it makes it easier to transfer project files to other computers.

Have a good day, ZB

**Reply by brunoherbelin** [reply]
I thought of this export feature for a long time and didn't have the incentive to implement it! Thanks for bringing this up !

Please try in the Beta the Export feature added for Playlists and for Session;

<img width="958" height="681" alt="image" src="https://github.com/user-attachments/assets/dea4c2a3-f10d-4b34-9d89-f9b98b3dd61e" />

What you mention as a 'Project' is what is called a Playlist in Vimix; that means a set of sessions.
Export allows to select one target directory (possibly external drive) and to copy all session files WITH their media dependencies. 
You can also of course export only one session (from session panel).

Thanks for testing; i didn't have time to do extensive tests with many files or large videos. I also didn't try all types of sources (e.g. Sequence) and didn't verify with Bundles or embedded sessions inside a session (but should work).

**Reply by zcbernard** [reply]
Hi bruno, 

Tested, it works fine. Thank you.
ZB

**Reply by zcbernard** [reply]
Hi, 
Retested, all options work. I haven't been able to test exporting to external drives because I can't allow access to media other than the Home folder, unlike the snaps version, even when using Flatseal.

ZB

**Reply by brunoherbelin** [reply]
Updated Release 0.9.1 should fix the Flatpak access ; thanks for pointing this out!

**Comment by lautriva** [comment]
Some ideas that popped in my head :
- Send the audio in the SRT stream
- Is it possible to add a "video sequence" source that let you play multiple videos one after the other (like a playlist)
  - At the moment I just concatenate them with ffmpeg
  - Not idea how it would look like in the player (One timeline with next/prev buttons?)
  - Not sure if it is still in Vimix scope?
